pages/docs/user-docs/build-docker-module.md

```diff
@@ -0,0 +1,43 @@
+---
+title: docker bootstrap module
+sidebar: user_docs
+permalink: build-docker-module
+folder: docs
+toc: false
+---
+
+This module allows you to build a container from an existing container hosted on Docker Hub or another Docker registry. 
+
+{% include toc.html %}
+
+## Overview
+You can use an existing container image as your "base," and then add customization. For example, instead of using the `debootstrap` module to start an Ubuntu container, you might use the official Ubuntu container from Docker Hub as a base.  You could do the same with CentOS, Debian, Arch, Suse, Alpine, BusyBox, etc.  Or maybe you want to build a container that uses CUDA and cuDNN to leverage the GPU, but ou don't want to install from scratch.  You can start with one of the `nvidia/cuda` containers and install your software on top of that.
+
+## Keywords
+```
+Bootstrap: docker
+```
+The **Bootstrap** keyword is always mandatory. It describes the bootstrap module to use.
+```
+From: /username/container:tag
+```
+The **From** keyword is mandatory.  It specifies the container to use as a base. In the case of some official registries (ubuntu for example) the `username` is optional.  The `tag` is also optional and will default to `latest`
+```
+Registry: http://custom_registry
+```
+The **Registry** keyword is optional.  It will default to `docker://index.docker.io`.
+```
+Token: yes
+```
+The **Token** keyword is optional.  It defaults to `no`.  In some cases the Docker API will request an authentication token which can be generated on the fly by using this keyword.  
+```
+IncludeCmd: yes
+```
+The **IncludeCmd** keyword is optional.  If included the Docker `ENTRYPOINT` or `CMD` will be used as a runscript.  Note that the IncludeCmd keyword is considered valid if it is _not empty_!  This means that `IncludeCmd: yes` and `IncludeCmd: no` are identical.  In both cases the IncludeCmd keyword is not empty, so the Docker `ENTRYPOINT` or `CMD` will be used for the runscript.  
+
```

Comment:
This isn't right either - see the docs I wrote about the order of operations for these things. Default we use entrypoint, they don't need to speciy anything. Then IncludeCmd says to use CMD instead, but those come second to a user provided runscript section.

Reply:
Yeah I read that section right before writing this but clearly I missed something.  😸 

So, if I understand correctly the point of `IncludeCmd` is to make `CMD` take precedence over `ENTRYPOINT`?  It basically has nothing to do with the `%runscript` section?  Is that right?

Reply:
Correct! Our idea of `%runscript` in Docker land is going to either be the ENTRYPOINT or CMD. If we specify our own runscript, then we don't care about either of the Docker commands (and the example would be building some image with a Docker base that has a /bin/sh entry but we have our custom scientific pipeline). However if we don't, then we would expect the image to be like Docker, so we treat ENTRYPOINT as runscript. But if the user wants to use the CMD instead, then they set IncludeCmd to yes. This was a very early addition that has lasted this long  - likely I would have made the functionality / name different, but no need to change now.